warrior/WarriorCore/common_execution_utils.py

```diff
@@ -12,8 +12,76 @@
 '''
 
 """Module that contains common utilities required for execution """
+import copy
+
 from Framework.Utils.print_Utils import print_warning
+import Framework.Utils as Utils
+
+def get_step_list(filepath, step_tag, sub_step_tag):
+    """
+    Takes the location of Testcase /Suite/Project xml file as input
+    finds a list of all the step elements present in the file
+    Parse the step object,
+    1. Retrieve the runmode and it corresponding value.
+    Based on runmode and value append the list.
+    2. Retrieve the retry and it corresponding value.
+    Based on runmode and value append the list.
+
+    :Arguments:
+        1. filepath    = full path of the Testcase/suite/project xml file
+        2. step_tag = xml tag for group of step in the file
+        3. sub_step_tag = xml tag for each step in the file
 
+    """
+    step_list_with_rmt_retry = []
+    root = Utils.xml_Utils.getRoot(filepath)
+    steps = root.find(step_tag)
+    if steps is None:
+        print_warning("The file: '{}' has no steps "
```

Comment:
can replaced the word steps with value of step_tag

Reply:
Done

Reply:
For step_list length > 1
there is a line
copy_step.find("runmode").set("value", go_next)
but it doesn't exist when step_list is empty

Reply:
Oops...What I mean is...step_tag is an xml element
and in the element there is a value specifying whether it is a step/testcase/testsuite
so the warning message can correctly say
The file X has no step/testcase/testsuite to be executed